File: 其他项目/nornir-vsr/backend/schemas/defaults.py
"""Defaults 配置模型。"""
from pydantic import BaseModel


class DefaultsUpdate(BaseModel):
    timeout: int
    global_delay_factor: float
    fast_cli: bool
    read_timeout: int
    num_workers: int
    license_module_enabled: bool


class DefaultsOut(DefaultsUpdate):
    pass

File: 其他项目/nornir-vsr/frontend/src/api/defaults.ts
import client from "./client";

export interface DefaultsConfig {
  timeout: number;
  global_delay_factor: number;
  fast_cli: boolean;
  read_timeout: number;
  num_workers: number;
  license_module_enabled: boolean;
}

export const fetchDefaults = async (): Promise<DefaultsConfig> => {
  const { data } = await client.get<DefaultsConfig>("/defaults");
  return data;
};

export const updateDefaults = async (payload: DefaultsConfig): Promise<DefaultsConfig> => {
  const { data } = await client.put<DefaultsConfig>("/defaults", payload);
  return data;
};
